packages/turf-distance/index.ts

```diff
@@ -1,13 +1,61 @@
 import { Point } from "geojson";
 import { getCoord } from "@turf/invariant";
-import { radiansToLength, degreesToRadians, Coord, Units } from "@turf/helpers";
+import {
+  radiansToLength,
+  degreesToRadians,
+  datums,
+  Coord,
+  Datum,
+  Units,
+  convertLength,
+} from "@turf/helpers";
+const LatLon = require("geodesy").LatLonEllipsoidal;
 
 //http://en.wikipedia.org/wiki/Haversine_formula
 //http://www.movable-type.co.uk/scripts/latlong.html
 
+/**
+ * Calculates the distance between two {@link Point|points}.
+ * If a specific datum is passed, uses a geodesic ellipsoid calculation.
+ * If no datum is passed, performs a great circle calculation.
+ *
+ * @name distance
+ * @param {Coord | Point} from origin point or coordinate
+ * @param {Coord | Point} to destination point or coordinate
+ * @param {Object} [options={}] Optional parameters
+ * @param {string} [options.units='kilometers'] can be degrees, radians, miles, or kilometers
+ * @param {Datum} [options.datum=datums.WGS84] datum listed in {@link Helpers}
+ * @returns {number} distance between the two points in chosen units
+ * @example
+ * var from = turf.point([-75.343, 39.984]);
+ * var to = turf.point([-75.534, 39.123]);
+ *
+ * var options = {units: 'miles'};
+ * var distance = turf.distance(from, to, options);
+ *
+ * //addToMap
+ * var addToMap = [from, to];
+ * from.properties.distance = distance;
+ * to.properties.distance = distance;
+ */
+function distance(
+  from: Coord | Point,
+  to: Coord | Point,
+  options: {
+    units?: Units;
+    datum?: Datum;
+  } = {}
```

Comment:
I'm seeing that this could become the standard way to pass a `datum` option to a turf function, if it supports it.  Seems like a good non-breaking start down this parth.